xarray/tests/test_dataarray.py

```diff
@@ -2909,6 +2909,13 @@ def test_to_and_from_dict(self):
                 ValueError, "cannot convert dict without the key 'data'"):
             DataArray.from_dict(d)
 
+        # check the data=False option
+        expected_no_data = {**expected}
```

Comment:
I guess this is a fancy Py3 way of doing a copy? :)

I would lean towards the more explicit `expected.copy()` given that you aren't inserting any extra fields here.